src/js/plugin.js

```diff
@@ -451,6 +451,26 @@ Plugin.BASE_PLUGIN_NAME = BASE_PLUGIN_NAME;
 
 Plugin.registerPlugin(BASE_PLUGIN_NAME, Plugin);
 
+/**
+ * Documented in player.js
+ *
+ * @ignore
+ */
+Player.prototype.usePlugin = function(name, plugin, ...options) {
+  let wrappedPlugin;
+
+  if (name !== BASE_PLUGIN_NAME) {
+    if (Plugin.isBasic(plugin)) {
+      wrappedPlugin = createBasicPlugin(name, plugin);
+    } else {
+      wrappedPlugin = createPluginFactory(name, plugin);
+    }
+    wrappedPlugin(...options);
+  } else {
+    throw new Error('Please help with nice error...');
+  }
```

Comment:
Given the necessary removal of the `plugin` argument, the body of this function can be simplified to:

```js
const plugin = getPlugin(name);

if (plugin) {
  plugin.apply(this, args);
} else {
  throw new Error(`Could not use plugin "${name}" as it does not exist.`);
}
```

Reply:
Requiring registration of plugin in separate function removes the most, if not all, of the reason to have it. The point is to have type anotations for plugin options and that is possible only, if they are in the same context.